Native.xs

```diff
@@ -111,7 +112,9 @@ void *DNS_getaddrinfo(void *v_arg) {
     if (arg->self->notify_on_begin)
         write(arg->res->fd1, "1", 1);
     arg->res->error = getaddrinfo(arg->host, arg->service, arg->hints, &arg->res->hostinfo);
-    
+    if (arg->res->error == EAI_SYSTEM)
+        arg->res->gai_errno = errno;
```

Comment:
Can we just reset `res->error` with `errno` in this case (if errno is not zero)? To simplify things.

Reply:
No, because then we can't know if we need to pass the value to `gai_strerror` or `strerror` later (see the `_get_result` chunk).

Reply:
Ah, I see. This naming `error` and `gai_errno` confuses. I think `gai_errno` should be `sys_error`, `error` may be renamed to `gai_error` or leaved as is. What do you think?

Reply:
I was thinking of `gai_errno` as "the `errno` value set by `g_a_i`", but if you think `sys_error` is clearer, I'm happy to change it.

Reply:
Yes, please

Reply:
Done.